src/frontend/src/stream_fragmenter/mod.rs

```diff
@@ -179,14 +193,27 @@ pub(self) fn build_and_add_fragment(
     state: &mut BuildFragmentGraphState,
     stream_node: StreamNode,
 ) -> Result<StreamFragment> {
-    let mut fragment = state.new_stream_fragment();
-    let node = build_fragment(state, &mut fragment, stream_node)?;
+    let operator_id = stream_node.operator_id as u32;
+    match state.share_mapping.get(&operator_id) {
+        None => {
+            let mut fragment = state.new_stream_fragment();
+            let node = build_fragment(state, &mut fragment, stream_node)?;
 
-    assert!(fragment.node.is_none());
-    fragment.node = Some(Box::new(node));
+            assert!(fragment.node.is_none());
+            fragment.node = Some(Box::new(node));
```

Comment:
Is it possible to make this an `Rc`?

Reply:
You mean make `build_and_add_fragment` return a `Result<Rc<StreamFragment>>` instead of `Result<StreamFragment>` ?

Reply:
Yes.